test/src/unittests/rhythm/test_percivalevaluatepulsetrains.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from essentia.standard import MonoLoader, PercivalEvaluatePulseTrains, HarmonicBpm, FrequencyBands, NoveltyCurve 
+
+class TestPercivalEvaluatePulseTrains(TestCase):
+
+    def testEmpty(self):        
+        lag = PercivalEvaluatePulseTrains()([],[])
+        self.assertEqual(-1.0, lag)
+
+    def testZero(self):
+        zeroOSS = zeros(1024)
+        zeropositions = zeros(1024)
+        lag = PercivalEvaluatePulseTrains()(zeroOSS, zeropositions)
+        self.assertEqual(0.0, lag)
+
+    def testConstantInput(self):
+        onesOSS = ones(1024)
+        onespositions = ones(1024)
+        lag = PercivalEvaluatePulseTrains()(onesOSS, onespositions)              
+        self.assertEqual(1.0, lag)
+    
+
+    def testRegression(self):
+        # Calculates the positions and Peaks
+        
+        inputSize=21168 # N.B The beat period is 21168 samples for 125 bpm @ 44.1k samp. rate
+        config = { 'range': inputSize -1, 'maxPosition': inputSize, 'minPosition': 0, 'orderBy': 'amplitude' }
```

Comment:
Any specific reason to use a config variable?  It complicates reading the code.

Reply:
I can remove it, but I though adding it would allow future extensions with more tests with varied configurations.